lib/interactive_viewer.ml

```diff
@@ -134,3 +141,14 @@ let view (patches : Patch.t list) =
     ]
 
 let start patch = Ui_loop.run ~quit ~tick_period:0.2 (view patch)
+let convert_char_to_key (c : char) : Ui.key = (`Uchar (Uchar.of_char c), [])
+
+let start_test patch events =
+  let content_ui = Lwd.quick_sample (Lwd.observe (view patch)) in
+  let ui_renderer = content_image_renderer content_ui in
+  let ui_events = List.map (fun e -> convert_char_to_key e) events in
+  List.iter
+    (fun event -> ignore (Renderer.dispatch_key ui_renderer event))
```

Comment:
A problem here is that you dispatch the key, but do not update the content of the renderer. Since we are not using the UI loop, we need to handle the changes of the Lwd elements by hand.
So for each dispatched event, you need to update the content of the renderer.